src/sentry/migrations/0742_backfill_broken_monitor_notification_setting_option.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 5.0.6 on 2024-07-17 08:41
+
+from django.apps.registry import Apps
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_broken_monitor_notification_setting_option(
+    apps: Apps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    NotificationSettingOption = apps.get_model("sentry", "NotificationSettingOption")
+
+    # Migrate all nudge settings to the new broken cron settings
+    approval_settings = NotificationSettingOption.objects.filter(type="approval")
```

Comment:
There isn't an index on the `type` column, so you'll be stuck doing a table scan which could timeout. Scanning all rows and doing this filtering in python will be slower, but give you a query that will not timeout.

Reply:
ahh makes sense tyty